Crash when invoked by xdg-desktop-portal

I keep seeing (somewhat rare, maybe once every few weeks) crashes when slurp is invoked by xdg-desktop-portal.
Backtrace:
```c
#0  0x00005696d20cb478 in send_frame (output=0x5696da9b84b0) at ../slurp-1.5.0/main.c:580
#1  0x00007102098a4ac6 in ffi_call_unix64 () at ../src/x86/unix64.S:104
#2  0x00007102098a176b in ffi_call_int (cif=cif@entry=0x7fff54c44230, fn=fn@entry=0x5696d20cbc10 <output_frame_handle_done>, rvalue=<optimized out>, rvalue@entry=0x0, 
    avalue=avalue@entry=0x7fff54c44300, closure=closure@entry=0x0) at ../src/x86/ffi64.c:676
#3  0x00007102098a406e in ffi_call (cif=cif@entry=0x7fff54c44230, fn=0x5696d20cbc10 <output_frame_handle_done>, rvalue=rvalue@entry=0x0, 
    avalue=avalue@entry=0x7fff54c44300) at ../src/x86/ffi64.c:713
#4  0x000071020a00048d in wl_closure_invoke (closure=closure@entry=0x5696da9baeb0, target=<optimized out>, target@entry=0x5696daa1b910, opcode=opcode@entry=0, 
    data=<optimized out>, flags=1) at ../wayland-1.24.0/src/connection.c:1241
#5  0x000071020a0012e9 in dispatch_event (display=display@entry=0x5696da9b2700, queue=queue@entry=0x5696da9b27f8) at ../wayland-1.24.0/src/wayland-client.c:1707
#6  0x000071020a0016f3 in dispatch_queue (display=0x5696da9b2700, queue=0x5696da9b27f8) at ../wayland-1.24.0/src/wayland-client.c:1853
#7  wl_display_dispatch_queue_pending (display=display@entry=0x5696da9b2700, queue=queue@entry=0x5696da9b27f8) at ../wayland-1.24.0/src/wayland-client.c:2190
#8  0x000071020a004f34 in wl_display_dispatch_queue_timeout (display=display@entry=0x5696da9b2700, queue=queue@entry=0x5696da9b27f8, timeout=timeout@entry=0x0)
    at ../wayland-1.24.0/src/wayland-client.c:2096
#9  0x000071020a005010 in wl_display_dispatch_queue (display=display@entry=0x5696da9b2700, queue=queue@entry=0x5696da9b27f8)
    at ../wayland-1.24.0/src/wayland-client.c:2163
#10 0x000071020a0050fc in wl_display_roundtrip_queue (display=0x5696da9b2700, queue=0x5696da9b27f8) at ../wayland-1.24.0/src/wayland-client.c:1459
#11 0x000071020a005151 in wl_display_roundtrip (display=<optimized out>) at ../wayland-1.24.0/src/wayland-client.c:1488
#12 0x00005696d20c98ea in main (argc=<optimized out>, argv=<optimized out>) at ../slurp-1.5.0/main.c:1093
```

with the following parameters:
```
Command Line: slurp -f %o -or
Executable: /usr/bin/slurp
Control Group: /user.slice/user-1000.slice/user@1000.service/app.slice/xdg-desktop-portal-wlr.service
Unit: user@1000.service
User Unit: xdg-desktop-portal-wlr.service
Slice: user-1000.slice
```
In gdb I get the crash here:
`output->current_buffer = get_next_buffer(state->shm, output->buffers, buffer_width, buffer_height);`
output->buffers is null, is that right?
```c
(gdb) p output->buffers
$8 = {{buffer = 0x0, surface = 0x0, cairo = 0x0, width = 0, height = 0, data = 0x0, size = 0, busy = false}, {buffer = 0x0, surface = 0x0, cairo = 0x0, width = 0, 
    height = 0, data = 0x0, size = 0, busy = false}}
(gdb) p *output
$11 = {wl_output = 0x5696da9b8, state = 0xad27dcee2271900, link = {prev = 0x0, next = 0x0}, geometry = {x = 0, y = 0, width = 1920, height = 1080, label = 0x0, link = {
      prev = 0x0, next = 0x0}}, logical_geometry = {x = 0, y = 0, width = 1920, height = 1080, label = 0x5696da9b28d0 "  \241ږV", link = {prev = 0x0, next = 0x0}}, 
  scale = 1, surface = 0x5696da9b75b0, layer_surface = 0x5696da9b85f0, xdg_output = 0x5696da9b8660, frame_callback = 0x0, configured = true, dirty = true, width = 1920, 
  height = 1080, buffers = {{buffer = 0x0, surface = 0x0, cairo = 0x0, width = 0, height = 0, data = 0x0, size = 0, busy = false}, {buffer = 0x0, surface = 0x0, 
      cairo = 0x0, width = 0, height = 0, data = 0x0, size = 0, busy = false}}, current_buffer = 0x5696da9b8590, cursor_theme = 0x5696da9b87b0, 
  cursor_image = 0x5696da9c4940}
(gdb) p *state
Cannot access memory at address 0xad27dcee2271900
```

```C
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[771]: 2025/10/04 19:42:17 [ERROR] - wlroots: no output found
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[771]: getline failed: Resource temporarily unavailable
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: ==32795==ABORTING
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Right alloca redzone:    cb
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Left alloca redzone:     ca
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   ASan internal:           fe
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Intra object redzone:    bb
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Array cookie:            ac
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Container overflow:      fc
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Poisoned by user:        f7
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Global init order:       f6
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Global redzone:          f9
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Stack use after scope:   f8
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Stack after return:      f5
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Stack right redzone:     f3
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Stack mid redzone:       f2
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Stack left redzone:      f1
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Freed heap region:       fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Heap left redzone:       fa
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Partially addressable: 01 02 03 04 05 06 07
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   Addressable:           00
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: Shadow byte legend (one shadow byte represents 8 application bytes):
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e2100: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e2080: fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa fa
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e2000: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e1f80: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e1f00: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fa fa
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: =>0x72dd423e1e80: fd fd fd fd fd fd fd fd fd fd[fd]fd fd fd fd fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e1e00: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e1d80: fd fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e1d00: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e1c80: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:   0x72dd423e1c00: fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa fa
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: Shadow bytes around the buggy address:
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: SUMMARY: AddressSanitizer: heap-use-after-free ../slurp-1.5.0/main.c:609 in output_frame_handle_done
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #4 0x75bd43c24ac5  (/usr/lib/libffi.so.8+0x7ac5) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #3 0x61f8f6877f8d in handle_global ../slurp-1.5.0/main.c:669
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #2 0x61f8f6877f8d in handle_global ../slurp-1.5.0/main.c:689
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #1 0x61f8f6877f8d in create_output ../slurp-1.5.0/main.c:534
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #0 0x75bd43f205dd in calloc /usr/src/debug/gcc/gcc/libsanitizer/asan/asan_malloc_linux.cpp:74
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: previously allocated by thread T0 here:
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #5 0x61f8f6874df4 in _start (/usr/bin/slurp+0x5df4) (BuildId: 781a1774dbd3aaf81304b9532500c1b30aa7aacd)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #4 0x75bd43a27728 in __libc_start_main (/usr/lib/libc.so.6+0x27728) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #3 0x75bd43a27674  (/usr/lib/libc.so.6+0x27674) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #2 0x61f8f6873b6d in main ../slurp-1.5.0/main.c:1085
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #1 0x61f8f6873b6d in destroy_output ../slurp-1.5.0/main.c:548
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #0 0x75bd43f1f79d in free /usr/src/debug/gcc/gcc/libsanitizer/asan/asan_malloc_linux.cpp:51
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: freed by thread T0 here:
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: 0x72dd423e1ed0 is located 144 bytes inside of 304-byte region [0x72dd423e1e40,0x72dd423e1f70)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #13 0x61f8f6874df4 in _start (/usr/bin/slurp+0x5df4) (BuildId: 781a1774dbd3aaf81304b9532500c1b30aa7aacd)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #12 0x75bd43a27728 in __libc_start_main (/usr/lib/libc.so.6+0x27728) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #11 0x75bd43a27674  (/usr/lib/libc.so.6+0x27674) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #10 0x61f8f6873d71 in main ../slurp-1.5.0/main.c:1093
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #9 0x75bd445e00fb in wl_display_roundtrip_queue (/usr/lib/libwayland-client.so.0+0x90fb) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #8 0x75bd445e000f in wl_display_dispatch_queue (/usr/lib/libwayland-client.so.0+0x900f) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #7 0x75bd445dff33 in wl_display_dispatch_queue_timeout (/usr/lib/libwayland-client.so.0+0x8f33) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #6 0x75bd445dc6f2 in wl_display_dispatch_queue_pending (/usr/lib/libwayland-client.so.0+0x56f2) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #5 0x75bd445dc2e8  (/usr/lib/libwayland-client.so.0+0x52e8) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #4 0x75bd445db48c  (/usr/lib/libwayland-client.so.0+0x448c) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #3 0x75bd43c2406d in ffi_call (/usr/lib/libffi.so.8+0x706d) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #2 0x75bd43c2176a  (/usr/lib/libffi.so.8+0x476a) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #1 0x75bd43c24ac5  (/usr/lib/libffi.so.8+0x7ac5) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]:     #0 0x61f8f687a458 in output_frame_handle_done ../slurp-1.5.0/main.c:609
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: WRITE of size 8 at 0x72dd423e1ed0 thread T0
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: ==32795==ERROR: AddressSanitizer: heap-use-after-free on address 0x72dd423e1ed0 at pc 0x61f8f687a459 bp 0x7ffe256920c0 sp 0x7ffe256920b0
Oct 04 19:42:17 arch xdg-desktop-portal-wlr[32795]: =================================================================
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[771]: 2025/10/04 19:42:14 [ERROR] - wlroots: no output found
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[771]: getline failed: Resource temporarily unavailable
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]: SUMMARY: AddressSanitizer: 96 byte(s) leaked in 1 allocation(s).
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]:     #6 0x7629b53dbac5  (/usr/lib/libffi.so.8+0x7ac5) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]:     #5 0x5b1e4c2868f9 in send_frame ../slurp-1.5.0/main.c:593
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]:     #4 0x5b1e4c2868f9 in wl_surface_frame /usr/include/wayland-client-protocol.h:4042
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]:     #3 0x7629b5df6fd2 in wl_proxy_marshal_flags (/usr/lib/libwayland-client.so.0+0x6fd2) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]:     #2 0x7629b5df5fad in wl_proxy_marshal_array_flags (/usr/lib/libwayland-client.so.0+0x5fad) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]:     #1 0x7629b5df5aed  (/usr/lib/libwayland-client.so.0+0x5aed) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]:     #0 0x7629b5f205dd in calloc /usr/src/debug/gcc/gcc/libsanitizer/asan/asan_malloc_linux.cpp:74
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]: Direct leak of 96 byte(s) in 1 object(s) allocated from:
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]: ==32788==ERROR: LeakSanitizer: detected memory leaks
Oct 04 19:42:14 arch xdg-desktop-portal-wlr[32788]: =================================================================
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[771]: 2025/10/04 19:41:51 [ERROR] - wlroots: no output found
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[771]: getline failed: Resource temporarily unavailable
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: ==32771==ABORTING
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Right alloca redzone:    cb
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Left alloca redzone:     ca
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   ASan internal:           fe
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Intra object redzone:    bb
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Array cookie:            ac
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Container overflow:      fc
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Poisoned by user:        f7
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Global init order:       f6
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Global redzone:          f9
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Stack use after scope:   f8
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Stack after return:      f5
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Stack right redzone:     f3
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Stack mid redzone:       f2
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Stack left redzone:      f1
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Freed heap region:       fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Heap left redzone:       fa
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Partially addressable: 01 02 03 04 05 06 07
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   Addressable:           00
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: Shadow byte legend (one shadow byte represents 8 application bytes):
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e2100: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e2080: fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa fa
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e2000: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e1f80: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e1f00: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fa fa
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: =>0x78ac9a1e1e80: fd fd fd fd fd fd fd fd fd fd[fd]fd fd fd fd fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e1e00: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e1d80: fd fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e1d00: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e1c80: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:   0x78ac9a1e1c00: fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa fa
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: Shadow bytes around the buggy address:
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: SUMMARY: AddressSanitizer: heap-use-after-free ../slurp-1.5.0/main.c:609 in output_frame_handle_done
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #4 0x7b8c9b6edac5  (/usr/lib/libffi.so.8+0x7ac5) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #3 0x5d79b3f94f8d in handle_global ../slurp-1.5.0/main.c:669
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #2 0x5d79b3f94f8d in handle_global ../slurp-1.5.0/main.c:689
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #1 0x5d79b3f94f8d in create_output ../slurp-1.5.0/main.c:534
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #0 0x7b8c9bd205dd in calloc /usr/src/debug/gcc/gcc/libsanitizer/asan/asan_malloc_linux.cpp:74
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: previously allocated by thread T0 here:
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #5 0x5d79b3f91df4 in _start (/usr/bin/slurp+0x5df4) (BuildId: 781a1774dbd3aaf81304b9532500c1b30aa7aacd)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #4 0x7b8c9b827728 in __libc_start_main (/usr/lib/libc.so.6+0x27728) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #3 0x7b8c9b827674  (/usr/lib/libc.so.6+0x27674) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #2 0x5d79b3f90b6d in main ../slurp-1.5.0/main.c:1085
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #1 0x5d79b3f90b6d in destroy_output ../slurp-1.5.0/main.c:548
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #0 0x7b8c9bd1f79d in free /usr/src/debug/gcc/gcc/libsanitizer/asan/asan_malloc_linux.cpp:51
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: freed by thread T0 here:
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: 0x78ac9a1e1ed0 is located 144 bytes inside of 304-byte region [0x78ac9a1e1e40,0x78ac9a1e1f70)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #13 0x5d79b3f91df4 in _start (/usr/bin/slurp+0x5df4) (BuildId: 781a1774dbd3aaf81304b9532500c1b30aa7aacd)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #12 0x7b8c9b827728 in __libc_start_main (/usr/lib/libc.so.6+0x27728) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #11 0x7b8c9b827674  (/usr/lib/libc.so.6+0x27674) (BuildId: 4fe011c94a88e8aeb6f2201b9eb369f42b4a1e9e)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #10 0x5d79b3f90d71 in main ../slurp-1.5.0/main.c:1093
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #9 0x7b8c9c3f60fb in wl_display_roundtrip_queue (/usr/lib/libwayland-client.so.0+0x90fb) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #8 0x7b8c9c3f600f in wl_display_dispatch_queue (/usr/lib/libwayland-client.so.0+0x900f) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #7 0x7b8c9c3f5f33 in wl_display_dispatch_queue_timeout (/usr/lib/libwayland-client.so.0+0x8f33) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #6 0x7b8c9c3f26f2 in wl_display_dispatch_queue_pending (/usr/lib/libwayland-client.so.0+0x56f2) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #5 0x7b8c9c3f22e8  (/usr/lib/libwayland-client.so.0+0x52e8) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #4 0x7b8c9c3f148c  (/usr/lib/libwayland-client.so.0+0x448c) (BuildId: 3b07142880be2014296cc8464e4410c69ed78d0d)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #3 0x7b8c9b6ed06d in ffi_call (/usr/lib/libffi.so.8+0x706d) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #2 0x7b8c9b6ea76a  (/usr/lib/libffi.so.8+0x476a) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #1 0x7b8c9b6edac5  (/usr/lib/libffi.so.8+0x7ac5) (BuildId: d5e3b0d8921923f35438adefa9f864745abc5e90)
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]:     #0 0x5d79b3f97458 in output_frame_handle_done ../slurp-1.5.0/main.c:609
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: WRITE of size 8 at 0x78ac9a1e1ed0 thread T0
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: ==32771==ERROR: AddressSanitizer: heap-use-after-free on address 0x78ac9a1e1ed0 at pc 0x5d79b3f97459 bp 0x7ffc2df4dfd0 sp 0x7ffc2df4dfc0
Oct 04 19:41:51 arch xdg-desktop-portal-wlr[32771]: =================================================================
```

--- a/main.c
+++ b/main.c
@@ -586,6 +586,9 @@ static void send_frame(struct slurp_output *output) {
 	render(output);
 
 	// Schedule a frame in case the output becomes dirty again
+	if (output->frame_callback) {
+		wl_callback_destroy(output->frame_callback);
+	}
 	output->frame_callback = wl_surface_frame(output->surface);
 	wl_callback_add_listener(output->frame_callback,
 		&output_frame_listener, output);

